docs/rules/no-async-before.md

```diff
@@ -0,0 +1,52 @@
+# Prevent using async/await in Cypress test cases (no-async-tests)
+
+Cypress before that return a promise will cause tests to prematurely start, can cause unexpected behavior. An `async` function returns a promise under the hood, so a before using an `async` function might also error.
```

Comment:
I wonder if there is a better way to phrase this? Can we find any concrete examples where before/beforeEach awaiting cypress commands cause an error similar to test bodies?
```suggestion
Cypress commands that return a promise may cause side effects in before/beforeEach hooks, possibly causing unexpected behavior. 
```